src/v/syschecks/syschecks.cc

```diff
@@ -48,24 +50,42 @@ void memory(bool ignore) {
     }
 }
 
-void systemd_notify_ready() {
+ss::future<> systemd_notify_ready() {
     ss::sstring msg = fmt::format(
       "READY=1\nSTATUS=redpanda is ready! - {}", redpanda_version());
-    systemd_raw_message(msg);
+    return systemd_raw_message(std::move(msg));
 }
 
-// TODO:agallego - support non-systemd installations
-// by adding a macro around sd_notify to disable at compile time
-void systemd_raw_message(const ss::sstring& out) {
-    auto r = sd_notify(0, out.c_str());
+ss::future<> systemd_raw_message(ss::sstring out) {
+    const char* systemd_socket_path = std::getenv("NOTIFY_SOCKET");
+    std::string_view log_msg = out;
     if (out.back() == '\n') {
-        checklog.debug("{}", std::string_view(out.c_str(), out.size() - 1));
-    } else {
-        checklog.debug("{}", out);
+        // emit tight logs without extra new lines
+        log_msg = std::string_view(out.c_str(), out.size() - 1);
+    }
+    if (!systemd_socket_path) {
+        checklog.trace("NOTIFY_SOCKET unset. ignoring {}", log_msg);
+        co_return;
+    }
+    checklog.debug("{}", log_msg);
+    ss::sstring systemd_socket = systemd_socket_path;
+    if (systemd_socket[0] == '@') {
+        // detected abstract socket; replace @ with 0
+        systemd_socket[0] = '0';
     }
-    if (unlikely(r < 0)) {
-        checklog.trace("Could not notify systemd sd_notify ready, error:{}", r);
+    auto sock = co_await ss::engine().net().connect(
+      ss::socket_address(ss::unix_domain_addr(systemd_socket)));
+    try {
+        auto output = sock.output();
+        co_await output.write(out);
+        co_await output.flush();
+        co_await output.close();
+    } catch (const std::exception& e) {
+        checklog.error("Error sending systemd notification: {}", e.what());
```

Comment:
Is the change from `trace` to `error` intended?

Reply:
yeah, this is an exception, before we just crashed since sd_notify is a C library.

Reply:
But now it's handled... so less of an error? Unless it's error level that it's not being sent anywhere more obvious.

Reply:
used the internal posix stack.